src/components/UserSelector/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from 'react';
+import { DropdownItem } from './DropdownItem';
+import { User } from '../../../types/User';
+
+interface DropdownMenuProps {
+  users: User[];
+  setIsActive: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+interface UserWithKey extends User {
+  key: string;
+}
+
+export const DropdownMenu: React.FC<DropdownMenuProps> = ({
+  users,
+  setIsActive,
+}) => {
+  const [usersWithKeys, setUsersWithKeys] = useState<UserWithKey[]>([]);
+
+  useEffect(() => {
+    const generateUniqueKeys = (userList: User[]): UserWithKey[] => {
+      return userList.map(user => ({
+        ...user,
+        key: `${user.id}-${Math.random().toString(36).substr(2, 9)}`,
```

Comment:
Generating keys using Math.random() is not recommended because it can lead to unstable keys. Instead, use a stable identifier like user.id directly as the key.